storage/redis/peer_store.go

```diff
@@ -84,34 +89,53 @@ func addKeyPrefix(namespacePrefix string, command string) string {
 	return command
 }
 
+func getIPType(ip interface{}) string {
+	switch v := ip.(type) {
+	case net.IP:
+		if len(v) == net.IPv6len {
+			return ipv6
+		} else {
+			return ipv4
+		}
+	case bool:
```

Comment:
instead of doing this, we could directly use `ipv4` and `upv6` strings in the ScrapeSwarm function like below
```
...
var ipType string = ipv4
if v6 {
  ipType = ipv6
}
```